File: projet_groupe4/src/main/java/projet_groupe4/dto/response/ClientWithEmpruntResponse.java
package projet_groupe4.dto.response;

import java.util.List;

import projet_groupe4.model.Client;

public class ClientWithEmpruntResponse {

    private Integer id;
    private String nom;
    private String prenom;
    private String email;
    private String telephone;
    private List<EmpruntResponse> emprunts;

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public String getTelephone() {
        return telephone;
    }

    public void setTelephone(String telephone) {
        this.telephone = telephone;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getNom() {
        return nom;
    }

    public void setNom(String nom) {
        this.nom = nom;
    }

    public String getPrenom() {
        return prenom;
    }

    public void setPrenom(String prenom) {
        this.prenom = prenom;
    }

    public List<EmpruntResponse> getEmprunts() {
        return emprunts;
    }

    public void setEmprunts(List<EmpruntResponse> emprunts) {
        this.emprunts = emprunts;
    }

    public static ClientWithEmpruntResponse convert(Client client) {
        ClientWithEmpruntResponse resp = new ClientWithEmpruntResponse();

        resp.setId(client.getId());
        resp.setNom(client.getNom());
        resp.setPrenom(client.getPrenom());
        resp.setEmail(client.getMail());
        resp.setTelephone(client.getTelephone());
        resp.setEmprunts(client.getEmprunts().stream().map(EmpruntResponse::convert).toList());

        return resp;
    }
}

File: projet_groupe4_angular/src/app/dto/client-with-emprunt-response.ts
import { Emprunt } from './emprunt';

export class ClientWithEmpruntResponse {
  constructor(
    private _id: number,
    private _nom: string,
    private _prenom: string,
    private _email: string,
    private _telephone: string,
    private _emprunts: Emprunt[]
  ) {}

  public get id(): number {
    return this._id;
  }

  public set id(value: number) {
    this._id = value;
  }

  public get nom(): string {
    return this._nom;
  }

  public set nom(value: string) {
    this._nom = value;
  }

  public get prenom(): string {
    return this._prenom;
  }

  public set prenom(value: string) {
    this._prenom = value;
  }

  public get email(): string {
    return this._email;
  }

  public set email(value: string) {
    this._email = value;
  }

  public get telephone(): string {
    return this._telephone;
  }

  public set telephone(value: string) {
    this._telephone = value;
  }

  public get emprunts(): Emprunt[] {
    return this._emprunts;
  }

  public set emprunts(value: Emprunt[]) {
    this._emprunts = value;
  }

  public toJson(): any {
    return {
      nom: this.nom,
      prenom: this.prenom,
      email: this.email,
      telephone: this.telephone,
      emprunts: this.emprunts,
    };
  }
}
